www/docs/operations.mdx

```diff
@@ -0,0 +1,248 @@
+beginning of an Effection program, but the easiest and most common way to
+run an operation is to include it in the body of another operation. To do this,
+we use the `yield*` keyword, which is the Effection equivalent of `await`.
+
+For example, the `sleep()` operation pauses execution for a specified duration
+before resuming. We can call it from any operation using `yield*`;
+
+```javascript
+import { main, sleep } from 'effection';
+
+await main(function*() {
+  yield* sleep(1000);
+  console.log("Hello World!");
+});
+```
+
+Or, it can be embeded in an operation which itself is embedded in another
+operation:
+
+```javascript
+import { main, sleep } from 'effection';
+
+function* makeSlow(value) {
+  yield* sleep(1000);
+  return value;
+}
+
+await main(function*() {
+  let text = yield* makeSlow('Hello World!');
+  console.log(text);
+});
+```
+
+There is no limit to the number of operations that can be nested within
+each other, and in fact, composition is so core to how operations work that
+_every operation in Effection eventually boils down to a combination of
+just three primitive operations_: `action()`, `resource()`, and `suspend()`.
+
+### Cancellation and Cleanup
+
+Perhaps the most critical difference between promises and operations is that
+once started, a promise will always run to completion no matter how long that
+takes. Unfortunately, this makes the guarantees of structured concurrency
+impossible to provide automatically. Operations, on the other hand, can be
+interrupted at any time which means that we never have to worry about them
+staying longer than they are absolutely needed.
+
+Let's consider a hypothetical sleep function implemented using `async/await`. It
+uses `setTimeout()` to resolve a promise once the sleep time has elapsed.
+
+```js
+async function sleep(duration) {
+  await new Promise(resolve => setTimeout(resolve, duration));
+}
+```
+
+We can now write our slow hello world program using this version:
+
+```js
+async function main() {
+  await sleep(1000);
+  console.log("Hello World!");
+}
+
+await main();
+```
+
+This logs `"Hello World!"` to the console after a delay of 1s, and so it works
+as expected when we run it... sort of.
+
+But what happens when you want to halt your sleep operation? Or more commonly,
+what happens when you no longer care about the outcome of your sleep operation?
+
+To understand why this might be, let's ask ourselves how long the program should
+take to execute in Node JS?
+
+```js
+async function sleep(milliseconds) {
+  await new Promise(resolve => setTimeout(resolve, milliseconds));
+}
+
+await Promise.race([sleep(10), sleep(10_000)]);
+```
+
+If you guessed 10 seconds, then you are technically correct! But if your
+intuition told you that the answer _should_ be 10 milliseconds, then
+congratulations, you are also correct!
+
+Our intuition tells us that the answer should be 10 milliseconds because when
```

Comment:
This paragraph doesn't tell them the answer, it kinda leaves them hanging until the last sentence and even then it doesn't mention the number. How about this?

```
It takes 1000ms even though our intuition says it should be 10ms. It takes 1000ms because `setTimeout` installs a callback and the run loop can not exist until that callback is removed. 

In practice, the application will be able to continue past the await statement after 10ms but your program will not finish for another 990ms until that callback is cleared.

By contast, the equivalent code in Effection will exit immidiately after 10ms as we expect.
```